### PR TITLE
ADEN-7353 Read consent from __cmp

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: src/services/tracking-opt-in.js [hunk 1]
```diff
@@ -4,9 +4,27 @@ import { queryString } from '../utils/query-string';
...
+		window.__cmp('getVendorConsents', null, (consents) => {
+			// we assume user opts in when he consents for at least one vendor
+			const optIn = Object.keys(consents.vendorConsents).filter(key => !!consents.vendorConsents[key]).length > 0;
+
+			context.set('options.trackingOptIn', isOptInByQueryParam || optIn);
+		});
+	}
+}
+
+readConsent();
```
Comment (fraszczakszymon) [inline src/services/tracking-opt-in.js]: Dear code reviewer: is it safe? It assumes that __cmp is loaded before ad-engine package. Should we readConset just before adEngine run?

Comment (fraszczakszymon) [reply src/services/tracking-opt-in.js]: Let me call that: https://github.com/Wikia/ad-engine/pull/152/commits/5d5e7b47f2118dcfb4bbd8ec9ec2f591c687b5e5

File: spec/services/tracking-opt-in.spec.js [hunk 2]
```diff
@@ -1,17 +1,56 @@
...
 import { context } from '../../src/services/context-service';
+import { trackingOptIn } from '../../src/services/tracking-opt-in';
+
+let vendorConsents = 50;
+
+function setupCmp() {
+	window.__cmp = (cmd, params, cb) => {
+		cb({
+			metadata: 'BOQu5naOQu5naCNABAAABRAAAAAAAA',
+			purposeConsents: Array.from({ length: 5 }).reduce((map, val, i) => {
```
Comment (jacekbj) [inline spec/services/tracking-opt-in.spec.js]: ditto

File: examples/utils/cmp/script.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		if (cmd === 'getConsentData') {
+			cb({
+				consentData: optIn ? 'BOQu5jyOQu5jyCNABAPLBR-AAAAeCAFgAUABYAIAAaABFACY' : 'BOQu5naOQu5naCNABAPLBRAAAAAeCAAA',
+				gdprApplies: true,
+				hasGlobalScope: false
+			}, true);
+		} else if (cmd === 'getVendorConsents') {
+			cb({
+				metadata: 'BOQu5naOQu5naCNABAAABRAAAAAAAA',
+				purposeConsents: Array.from({ length: 5 }).reduce((map, val, i) => {
```
[the remark this answers is not in the document]
Comment (jacekbj) [reply examples/utils/cmp/script.js]: Apart from that, the trick with `reduce` is very neat ;)

File: examples/utils/cmp/script.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+context.set('options.trackingOptIn', null);
+
+const isCmpDefined = typeof utils.queryString.get('cmp') !== 'undefined',
+	optIn = utils.queryString.get('cmp') === '1';
+
+if (isCmpDefined) {
+	window.__cmp = function (cmd, param, cb) {
+		if (cmd === 'getConsentData') {
+			cb({
+				consentData: optIn ? 'BOQu5jyOQu5jyCNABAPLBR-AAAAeCAFgAUABYAIAAaABFACY' : 'BOQu5naOQu5naCNABAPLBRAAAAAeCAAA',
```
[the remark this answers is not in the document]
Comment (leshniak) [reply examples/utils/cmp/script.js]: @jacekbj the whole file is just a mock ;)

**Pull request closed without merging**